src/Features/VisualBasic/Portable/RemoveUnnecessaryImports/VisualBasicRemoveUnnecessaryImportsService.vb

```diff
@@ -33,9 +32,19 @@ Namespace Microsoft.CodeAnalysis.VisualBasic.RemoveUnnecessaryImports
                     End Function).ToSet()
         End Function
 
-        Public Function RemoveUnnecessaryImports(document As Document, model As SemanticModel, root As SyntaxNode, cancellationToken As CancellationToken) As Document Implements IRemoveUnnecessaryImportsService.RemoveUnnecessaryImports
+        Public Async Function RemoveUnnecessaryImportsAsync(document As Document, cancellationToken As CancellationToken) As Task(Of Document) Implements IRemoveUnnecessaryImportsService.RemoveUnnecessaryImportsAsync
             Using Logger.LogBlock(FunctionId.Refactoring_RemoveUnnecessaryImports_VisualBasic, cancellationToken)
-                Dim unnecessaryImports = DirectCast(GetIndividualUnnecessaryImports(model, root, cancellationToken), ISet(Of ImportsClauseSyntax))
+                Dim model = Await document.GetSemanticModelAsync(cancellationToken).ConfigureAwait(False)
+                Dim root = Await document.GetSyntaxRootAsync(cancellationToken).ConfigureAwait(False)
+
+                Dim unnecessaryImports = New HashSet(Of ImportsClauseSyntax)(GetIndividualUnnecessaryImportsOrEmpty(model, root, cancellationToken), Me)
```

Comment:
This code looks identical.  It feels like the VBRemoveUnnecessaryService and the CSharpRemoveUnnecessaryService should share an abstract base class to remove duplicated logic that we have to keep in sync.